scalikejdbc-core/src/main/scala/scalikejdbc/DBConnection.scala

```diff
@@ -461,18 +463,39 @@ trait DBConnection extends LogSupport with LoanPattern {
               new RSTraversable(meta.getIndexInfo(null, schema, table, false, true))
                 .foldLeft(Map[String, Index]()) {
                   case (map, rs) =>
-                    val indexName = rs.string("INDEX_NAME")
-                    val index = map.get(indexName).map { index =>
-                      index.copy(columnNames = rs.string("COLUMN_NAME") :: index.columnNames)
-                    }.getOrElse {
-                      Index(
-                        name = indexName,
-                        columnNames = List(rs.string("COLUMN_NAME")),
-                        isUnique = !rs.boolean("NON_UNIQUE"))
+                    val indexName: String = rs.string("INDEX_NAME")
+                    val index: Index = map.get(indexName) match {
+                      case Some(idx) =>
+                        rs.stringOpt("COLUMN_NAME") match {
+                          case Some(columnName) => idx.copy(columnNames = idx.columnNames :+ columnName)
+                          case _ => idx
+                        }
+                      case _ =>
+                        Index(
+                          name = indexName,
+                          columnNames = rs.stringOpt("COLUMN_NAME").toList,
+                          isUnique = !rs.boolean("NON_UNIQUE"),
+                          qualifier = rs.stringOpt("INDEX_QUALIFIER"),
+                          indexType = {
+                            rs.shortOpt("TYPE") match {
+                              case Some(t) => IndexType.from(t)
+                              case _ => IndexType.tableIndexOther
```

Comment:
I'm not sure choosing tableIndexOther is the best when null comes from JDBC drivers. IMO it's reasonable enough though.